packages/flutter_tools/templates/app_shared/windows.tmpl/runner/Runner.rc.tmpl

```diff
@@ -60,21 +60,31 @@ IDI_APP_ICON            ICON                    "resources\\app_icon.ico"
 // Version
 //
 
-#ifdef FLUTTER_BUILD_NUMBER
-#define VERSION_AS_NUMBER FLUTTER_BUILD_NUMBER
+// The extra level of indirection is necessary to make the preprocessor expand
+// the macro to the token's value instead of the token itself.
+// See: https://stackoverflow.com/a/5459929
+#define _TOSTRING(x) #x
+#define TOSTRING(x) _TOSTRING(x)
+
+#if defined(FLUTTER_BUILD_MAJOR) && defined(FLUTTER_BUILD_MINOR) && defined(FLUTTER_BUILD_PATCH) && defined(FLUTTER_BUILD_NUMBER)
+#define FILE_VERSION_AS_NUMBER FLUTTER_BUILD_MAJOR,FLUTTER_BUILD_MINOR,FLUTTER_BUILD_PATCH,FLUTTER_BUILD_NUMBER
+#define PRODUCT_VERSION_AS_NUMBER FLUTTER_BUILD_MAJOR,FLUTTER_BUILD_MINOR,FLUTTER_BUILD_PATCH
 #else
-#define VERSION_AS_NUMBER 1,0,0
+#define FILE_VERSION_AS_NUMBER 1,0,0,1
+#define PRODUCT_VERSION_AS_NUMBER 1,0,0,
 #endif
 
-#ifdef FLUTTER_BUILD_NAME
-#define VERSION_AS_STRING #FLUTTER_BUILD_NAME
```

Comment:
If the old code that was intended to future-proof existing projects was wrong, then you'll need to either write a migrator (which would be better) or use different defines so that the old code doesn't trigger, otherwise existing projects will get garbage version information.

Reply:
As per the pull request description, existing projects aren't affected by this change until they "upgrade" their project by recreating their `windows` folder. In other words, folks that that do not regenerate their `windows` folder will have version `1.0.0` even if they upgrade their Flutter SDK. 

You need to update your your `windows/runner/CMakeLists.txt` file before the version info flows to your Windows build. This file is what bridges the generated CMake config to your `Runner.rc` file.

Given customers aren't broken, @cbracken and I feel it is acceptable for us to add a `flutter fix` in a subsequent PR. Please let me know if you disagree or have additional feedback!

Reply:
What is the mechanism that will cause existing projects—which have ifdefs based on `FLUTTER_BUILD_NUMBER`—to be unaffected by the fact that the tool will now define `FLUTTER_BUILD_NUMBER`?

Reply:
Never mind, I missed that they layer to turn the Flutter definitions into compile definitions is in the CMake file.

Reply:
Nothing is flowing `FLUTTER_BUILD_NUMBER` or `FLUTTER_BUILD_NAME` to your `Runner.rc` file today.

The new version info flow is:

1. Flutter SDK reads version info from build arguments and your `pubspec.yaml` 
2. **New**: Flutter SDK passes version info to CMake using ephemeral generated config file
3. **New**: CMake passes version info to Windows build using `/windows/runner/CMakeList.txt` file
4. **New**: `Runner.rc` stamps the version info to your build

If you upgrade your Flutter SDK, you'll get steps 1 and 2... but since step 3 is missing, your existing `Runner.rc` file won't get the version info.

By the way, the current `Runner.rc` file is broken. Today, you will get a build error if your build defines `FLUTTER_BUILD_NAME`.